### PR TITLE
enable s390x to test containers

### PR DESCRIPTION
**Pull request by liudalibj**
What type of PR is this?
/kind failing-test

What this PR does / why we need it:
Conformance testing on s390x https://github.com/kubernetes/kubernetes/issues/67721

Which issue(s) this PR fixes:

Fixes #
conformance failures on s390x

Special notes for your reviewer:

This PR is same as https://github.com/kubernetes/kubernetes/pull/75344, 
build out images: google-samples/gb-frontend-s390x:v6 and google-samples/gb-redisslave-s390x:v3

### CODE REVIEW

**Comment by liudalibj** [comment]
I have got the CNCF, not sure why doesn't pass the CLA check.
Screen Shot 2019-05-13 at 5 54 13 PM

**Comment by liudalibj** [comment]
@mkumatag would you like help to review too.

File: guestbook/php-redis/Dockerfile [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
 FROM php:5-apache
 
+COPY fixup-apt-list.sh /
```
Comment (mkumatag) [inline guestbook/php-redis/Dockerfile]: I understand we don't have security updates for jessie release, but I'm wondering whether we can update to the latest version which has stretch as base image which does work without any of these crazy patches!

Comment (liudalibj) [reply guestbook/php-redis/Dockerfile]: the image can work well without these crazy patchers.
remove the update command.
test AMD64:
```
root@zkubetest:~# docker run gcr.io/google-samples/gb-frontend:v6
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
[Mon May 13 15:20:09.977625 2019] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.25 (Debian) PHP/5.6.40 configured -- resuming normal operations
[Mon May 13 15:20:09.977691 2019] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
^C[Mon May 13 15:20:12.871811 2019] [mpm_prefork:notice] [pid 1] AH00169: caught SIGTERM, shutting down
root@zkubetest:~#
```
test s390x:
```
root@d230718d690f:/home/liudali# docker run gcr.io/google-samples/gb-frontend-s390x:v6
AH00557: apache2: apr_sockaddr_info_get() failed for c3a219b07c49
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
AH00557: apache2: apr_sockaddr_info_get() failed for c3a219b07c49
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
[Mon May 13 15:15:24.111847 2019] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.10 (Debian) PHP/5.6.32 configured -- resuming normal operations
[Mon May 13 15:15:24.111879 2019] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
^C[Mon May 13 15:15:30.170413 2019] [mpm_prefork:notice] [pid 1] AH00169: caught SIGTERM, shutting down
root@d230718d690f:/home/liudali#
```
@mkumatag please review again.

**Comment by liudalibj** [comment]
PR is ready for approve and merge

**Comment by liudalibj** [comment]
@idvoretskyi @sebgoa would you like help to review this PR, this one is helpful to run conformance test base on kubernetes cluster which one have Z worker nodes.
With this pr it will build out s390x image for gcr.io/google-samples/gb-redisslave:v3 and gcr.io/google-samples/gb-frontend:v6

**Comment by liudalibj** [comment]
related codes in e2e
https://github.com/kubernetes/kubernetes/blob/03232736ba33b95a5e98144e82c1dbe77b427c36/test/e2e/common/util.go#L87
https://github.com/kubernetes/kubernetes/blob/03232736ba33b95a5e98144e82c1dbe77b427c36/test/e2e/common/util.go#L88
https://github.com/kubernetes/kubernetes/blob/d881c0d77bef9c268433c6ec0770fcc80ab79c59/test/e2e/kubectl/kubectl.go#L325

**Comment by mkumatag** [comment]
@liudalibj we need to ask someone who has access to gcr repo to push the images as well.!

**Comment by liudalibj** [comment]
@idvoretskyi thank you for approve and merge this PR.
@mkumatag got it thank you point out this.
@ixdy @listx would you like help to build and push the s390x image for gcr.io/google-samples/gb-redisslave:v3 and gcr.io/google-samples/gb-frontend:v6 to gcr repo? 
follow command would be helpful:
```
cd ~/git
git clone git@github.com:kubernetes/examples.git
cd ~/git/examples/guestbook/php-redis
make all
make all-push
cd ~/git/examples/guestbook/redis-slave
make all
make all-push
```
I would like to provide any help on this and I have verified that these images can be built out on Ubuntu 18.04.2 and these s390x images works as expected on Z platform.

**Comment by listx** [comment]
Hmm I actually do not have push access to that repo. @ixdy, do you have access?

**Comment by ixdy** [comment]
I think you need to reach out to @ahmetb.

**Comment by ahmetb** [comment]
@ixdy I might give you access for this, I am no longer the maintainer of the gcr.io/gogole-samples.

@dansanche if you have time, do you mind running the build/push command on the examples/guestbook sample?

**Comment by liudalibj** [comment]
thanks for your help @ixdy @ahmetb @listx @DanSanche @dims 
@DanSanche would you like update here when you finish the push command?

**Pull request merged**